sklearn/datasets/lfw.py

```diff
@@ -72,36 +89,37 @@ def check_fetch_lfw(data_home=None, funneled=True, download_if_missing=True):
     lfw_home = join(data_home, "lfw_home")
 
     if funneled:
-        archive_path = join(lfw_home, FUNNELED_ARCHIVE_NAME)
         data_folder_path = join(lfw_home, "lfw_funneled")
-        archive_url = BASE_URL + FUNNELED_ARCHIVE_NAME
+        archive_path = join(lfw_home, FUNNELED_ARCHIVE.filename)
+        archive_url = FUNNELED_ARCHIVE.url
+        expected_archive_checksum = FUNNELED_ARCHIVE.checksum
     else:
-        archive_path = join(lfw_home, ARCHIVE_NAME)
         data_folder_path = join(lfw_home, "lfw")
-        archive_url = BASE_URL + ARCHIVE_NAME
+        archive_path = join(lfw_home, ARCHIVE.filename)
+        archive_url = ARCHIVE.url
+        expected_archive_checksum = ARCHIVE.checksum
 
     if not exists(lfw_home):
         makedirs(lfw_home)
 
-    for target_filename in TARGET_FILENAMES:
-        target_filepath = join(lfw_home, target_filename)
+    for target in TARGETS:
+        target_filepath = join(lfw_home, target.filename)
         if not exists(target_filepath):
             if download_if_missing:
-                url = BASE_URL + target_filename
-                logger.warning("Downloading LFW metadata: %s", url)
-                urllib.urlretrieve(url, target_filepath)
+                logger.warning("Downloading LFW metadata: %s", target.url)
+                _fetch_url(target.url, target_filepath, target.checksum)
             else:
                 raise IOError("%s is missing" % target_filepath)
 
     if not exists(data_folder_path):
 
         if not exists(archive_path):
             if download_if_missing:
-                archive_path_temp = archive_path + ".tmp"
                 logger.warning("Downloading LFW data (~200MB): %s",
                                archive_url)
-                urllib.urlretrieve(archive_url, archive_path_temp)
-                rename(archive_path_temp, archive_path)
+
+                _fetch_url(archive_url, archive_path,
+                           expected_archive_checksum)
             else:
                 raise IOError("%s is missing" % target_filepath)
```

Comment:
I think this is wrong. It should be raising with regards to `archive_path` not target.